### ISSUE TITLE
@typescript-eslint/indent incorrect when using leading semicolon

### ISSUE THREAD

**Issue by nbilyk** [eslint/eslint#16661]
### Environment

Node version: 14.20
npm version: 6.14
Local ESLint version: 7.32.0
Operating System: MacOS

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
"semi": ["error", "never"],
"@typescript-eslint/indent": [
      "error", 4, {
        "ignoredNodes": [
          "TSTypeParameterInstantiation"
        ],
        "SwitchCase": 1
      }],
```
</details>

When the semi rule is set to never, there is one case where a property must begin with a semicolon: when using the index operator. 

```js
class A {

    propA = 1

    // es-lint extra indentation:
        ;[symbol1] = 2
}
```

### What did you expect to happen?

I expect a statement with a leading semicolon not to increase the necessary indentation

```js
class A {

    propA = 1

    ;[symbol1] = 2

}
```

### What actually happened?

```js
class A {

    propA = 1

        ;[symbol1] = 2

}
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nbilyk** [comment]
ah, thanks for the fast response. I like the big all caps, "this rule is broken"
I've heard that whitespace formatting is the most difficult problem of any IDE language plugin. I believe it.
